src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table-key-value/table-key-value.component.spec.ts

```diff
@@ -85,136 +83,90 @@ describe('TableKeyValueComponent', () => {
       { key: 'dash', value: 'board' }
     ]);
-    expect(component._makePairsFromObject({ dash: 'board', ceph: 'mimic' })).toEqual([
+    expect(component.makePairsFromObject({ dash: 'board', ceph: 'mimic' })).toEqual([
       { key: 'dash', value: 'board' },
       { key: 'ceph', value: 'mimic' }
     ]);
   });
 
-  describe('tests _convertValue', () => {
-    const v = (value) => ({ key: 'sth', value: value });
-    const testConvertValue = (value, result) =>
-      expect(component._convertValue(v(value)).value).toBe(result);
+  describe('tests convertValue()', () => {
+    const expectConvertValue = (value, expectation) =>
+      expect(component._convertValue(value)).toBe(expectation);
 
-    it('should leave a string as it is', () => {
-      testConvertValue('something', 'something');
+    it('should not convert strings', () => {
+      expectConvertValue('something', 'something');
     });
 
-    it('should leave an int as it is', () => {
-      testConvertValue(29, 29);
+    it('should not convert integers', () => {
+      expectConvertValue(29, 29);
     });
 
-    it('should convert arrays with any type to string', () => {
-      testConvertValue([1, 2, 3], '1, 2, 3');
-      testConvertValue([{ sth: 'something' }], '{"sth":"something"}');
-      testConvertValue([1, 'two', { 3: 'three' }], '1, two, {"3":"three"}');
+    it('should convert arrays with any type to strings', () => {
+      expectConvertValue([1, 2, 3], '1, 2, 3');
+      expectConvertValue([{ sth: 'something' }], '{"sth":"something"}');
+      expectConvertValue([1, 'two', { 3: 'three' }], '1, two, {"3":"three"}');
     });
 
-    it('should convert only allow objects if renderObjects is set to true', () => {
-      expect(component._convertValue(v({ sth: 'something' }))).toBe(undefined);
+    it('should only convert objects if renderObjects is set to true', () => {
+      expect(component._convertValue({ sth: 'something' })).toBe(null);
       component.renderObjects = true;
-      expect(component._convertValue(v({ sth: 'something' }))).toEqual(v({ sth: 'something' }));
+      expect(component._convertValue({ sth: 'something' })).toEqual({ sth: 'something' });
     });
 
-    describe('automatically pipe utc dates through cdDate', () => {
+    describe('automatically pipe UTC dates through cdDate', () => {
       let datePipe: CdDatePipe;
 
       beforeEach(() => {
         datePipe = TestBed.get(CdDatePipe);
         spyOn(datePipe, 'transform').and.callThrough();
       });
 
-      const expectTimeConversion = (date: string) => {
-        component.data = {
-          3: 'some time',
-          someKey: date
-        };
+      const expectTimeConversion = (date: string, expectation: string) => {
+        component.data = { dateKey: date };
         component.ngOnInit();
-        expect(component.tableData.length).toBe(2);
         expect(datePipe.transform).toHaveBeenCalledWith(date);
-        expect(component.tableData[1].key).not.toBe(date);
```

Comment:
Use this expectation, as you will run into time zone and time format issues if you try to expect a certain date and format.

That's the reason make check fails:

```
expect(received).toEqual(expected) // deep equality

    - Expected
    + Received

      Array [
        Object {
          "key": "dateKey",
    -     "value": "4/16/19 2:35:46 PM",
    +     "value": "4/16/19 12:35:46 PM",
        },
      ]

      159 |         component.ngOnInit();
      160 |         expect(datePipe.transform).toHaveBeenCalledWith(date);
    > 161 |         expect(component.tableData).toEqual([{ key: 'dateKey', value: expectation }]);
          |                                     ^
      162 |       };
```